### PR TITLE
Add compiler messages to semanticdb

### PR DESCRIPTION
**Pull request by olafurpg**
Follows #748

### CODE REVIEW

File: scalahost/src/main/protobuf/semanticdb.proto [hunk 1]
```diff
@@ -16,6 +16,7 @@ message Database {
 message DatabaseFile {
   Address address = 1;
   repeated SymbolRef symbols = 2;
+  repeated CompilerMessage messages = 3;
```
Comment (xeno-by) [inline scalahost/src/main/protobuf/semanticdb.proto]: \o/ I'm very excited to see how we can explore custom sections in the future!

File: scalahost/src/main/scala/scala/meta/internal/scalahost/HijackReporter.scala [hunk 2]
```diff
@@ -0,0 +1,32 @@
+package scala.meta.internal.scalahost
+
+import scala.meta.semantic.v1.CompilerMessage
+import scala.reflect.internal.util.Position
+import scala.tools.nsc.reporters.Reporter
+import scala.tools.nsc.reporters.StoreReporter
+
+import org.scalameta.logger
```
Comment (xeno-by) [inline scalahost/src/main/scala/scala/meta/internal/scalahost/HijackReporter.scala]: Why do we need this guy?

File: scalahost/src/main/scala/scala/meta/internal/scalahost/v1/online/DatabaseOps.scala [hunk 3]
```diff
@@ -8,9 +8,11 @@ import scala.reflect.internal.util._
 import scala.reflect.internal.Flags._
 import scala.tools.nsc.Global
 import scala.{meta => m}
-import scala.meta.semantic.v1.{Location, Database}
+import scala.meta.semantic.v1.{Database, Location}
 import scala.compat.Platform.EOL
 
+import org.scalameta.logger
```
Comment (xeno-by) [inline scalahost/src/main/scala/scala/meta/internal/scalahost/v1/online/DatabaseOps.scala]: Is this necessary?

File: scalameta/semantic/src/main/scala/scala/meta/semantic/v1/Database.scala [hunk 4]
```diff
@@ -13,20 +13,25 @@ import scala.meta.semantic.v1._
...
       buf ++= (addr + EOL)
       val content = addr.content
-      grouped(addr).keys.toList.sortBy(_.start).foreach(k => {
-        val snippet = content.substring(k.start, k.end)
-        buf ++= (s"[${k.start}..${k.end}): $snippet => ${grouped(addr)(k).syntax}" + EOL)
-      })
+      combined.foreach { case (Location(_, start, end), syntax) =>
+        val snippet = content.substring(start, end)
+        buf ++= (s"[$start..$end): $snippet => $syntax" + EOL)
+      }
```
Comment (xeno-by) [inline scalameta/semantic/src/main/scala/scala/meta/semantic/v1/Database.scala]: Very nice! I like the unification.

File: scalahost/src/main/scala/scala/meta/internal/scalahost/HijackReporter.scala [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+
+/** Reporter that forwards messages to underlying reporter AND stores messages. */
+class ScalahostReporter(underlying: Reporter) extends StoreReporter {
```
Comment (xeno-by) [inline scalahost/src/main/scala/scala/meta/internal/scalahost/HijackReporter.scala]: Is this used anywhere?

Comment (olafurpg) [inline scalahost/src/main/scala/scala/meta/internal/scalahost/HijackReporter.scala]: Right below https://github.com/scalameta/scalameta/pull/749/files#diff-0c0952dd1250709be3cf8e79097fe7d9R27

Comment (xeno-by) [inline scalahost/src/main/scala/scala/meta/internal/scalahost/HijackReporter.scala]: Sure, but where do we make use of it?

upd. Aaahhh, I see now. `CompilationUnit.messages` is an extension method. I could bet that there was such method already in scalac.

Comment (olafurpg) [inline scalahost/src/main/scala/scala/meta/internal/scalahost/HijackReporter.scala]: I renamed `messages` to `reportedMessages`

**Pull request merged**